src/hal/components/limit3.comp

```diff
@@ -2,94 +2,118 @@ component limit3 "Limit the output signal to fall between min and max, limit its
 pin in float in;
 pin out float out;
 pin in bit load "When TRUE, immediately set \\fBout\\fB to \\fBin\\fR, ignoring maxv and maxa";
-pin in float min_=-1e20;
-pin in float max_=1e20;
+pin in float min=-1e20;
+pin in float max=1e20;
 pin in float maxv=1e20;
 pin in float maxa=1e20;
-option data limit3_data;
+variable double in_pos_old;
+variable double out_vel_old;
 function _;
 license "GPL";
 ;;
 
 #include "rtapi_math.h"
 
-typedef struct {
-    double old_in;	/* previous input */
-    double old_out;	/* previous output */
-    double old_v;	/* previous 1st derivative */
-} limit3_data;
+#define SET_NEXT_STATE(out_pos, out_vel, in_pos)	\
```

Comment:
Nobody likes complicated macros like this (well, I kinda do), but it's a necessary evil because you can't pass a pin to a function using convenient syntax

Reply:
It's also handy to be able to `return;` from within the macro.  IMO it makes the rest of the logic a lot cleaner.